sklearn/feature_selection/rfe.py

```diff
@@ -146,6 +149,13 @@ def fit(self, X, y):
 
         support_ = np.ones(n_features, dtype=np.bool)
         ranking_ = np.ones(n_features, dtype=np.int)
+
+        if step_score:
+            # Determine the number of subsets of features
+            self.scores_ = np.zeros(
```

Comment:
I am not saying that it is complicated, I'm saying that you could have fewer lines do the same. You have to compute indices again in ``_do_step_score`` actually you could probably eliminate this function entirely if you used a list, right? You can later convert the list to an array.
I'd just like to have as few indirections and as little code as possible, as that will make it easier for someone to follow the code later.